core/common/src/test/java/alluxio/heartbeat/HeartbeatThreadTest.java

```diff
@@ -87,49 +92,48 @@ public void serialHeartbeatThreadTest() throws Exception {
    */
   @Test
   public void concurrentHeartbeatThreadTest() throws Exception {
-    List<Thread> mThreads = new LinkedList<>();
+    List<FutureTask<Null>> tasks = new LinkedList<>();
 
     // Start the threads.
     for (int i = 0; i < NUMBER_OF_THREADS; i++) {
-      Thread thread = new DummyHeartbeatTestThread(i);
+      FutureTask<Null> task = new FutureTask<>(new DummyHeartbeatTestCallable(i));
+      Thread thread = new Thread(task);
       thread.start();
-      mThreads.add(thread);
+      tasks.add(task);
     }
 
     // Wait for the threads to finish.
-    for (Thread thread : mThreads) {
-      thread.join();
+    for (FutureTask<Null> task: tasks) {
+      task.get();
     }
   }
 
-  private class DummyHeartbeatTestThread extends Thread  {
-    private String mThreadName;
+  private class DummyHeartbeatTestCallable implements Callable<Null>  {
+    private final String mThreadName;
 
-    public DummyHeartbeatTestThread() {
+    public DummyHeartbeatTestCallable() {
       mThreadName = THREAD_NAME;
     }
 
-    public DummyHeartbeatTestThread(int id) {
+    public DummyHeartbeatTestCallable(int id) {
       mThreadName = THREAD_NAME + "-" + id;
     }
 
     @Override
-    public void run()  {
-      try {
-        Whitebox.invokeMethod(HeartbeatContext.class, "setTimerClass", mThreadName,
-            HeartbeatContext.SCHEDULED_TIMER_CLASS);
-
+    public Null call() throws Exception {
+      try (ManuallyScheduleHeartbeat.Resource r =
+          new ManuallyScheduleHeartbeat.Resource(Arrays.asList(mThreadName))) {
         DummyHeartbeatExecutor executor = new DummyHeartbeatExecutor();
         HeartbeatThread ht = new HeartbeatThread(mThreadName, executor, 1);
 
         // Run the HeartbeatThread.
-        sExecutorService.submit(ht);
+        mExecutorService.submit(ht);
 
         // Wait for the DummyHeartbeatExecutor executor to be ready to execute its heartbeat.
-        Assert.assertTrue("Initial wait failed.",
+        Assert.assertTrue("Initial wait failed for " + mThreadName,
             HeartbeatScheduler.await(mThreadName, 5, TimeUnit.SECONDS));
 
-        final int numIterations = 100000;
+        final int numIterations = 200;
```

Comment:
The test takes about 1 second per 10k iterations. I've changed it to 5k iterations now. I don't think this test is worth adding a significant amount of time to the build